### PR TITLE
Add need_repeat_flag in phone based ctc graph compiler

### PR DESCRIPTION
**Pull request by pkufool**

### CODE REVIEW

File: icefall/graph_compiler.py [hunk 1]
```diff
@@ -29,6 +29,7 @@ def __init__(
         lexicon: Lexicon,
         device: torch.device,
         oov: str = "<UNK>",
+        need_repeat_flag: bool = False,
```
Comment (csukuangfj) [inline icefall/graph_compiler.py:32]: You may also need to modify the corresponding `train.py` to pass True to this funciton.

Comment (pkufool) [reply icefall/graph_compiler.py:32]: I think we only have bpe models that support delay-penalty, @yaozengwei  Did you update the phone-based recipes?

Comment (csukuangfj) [reply icefall/graph_compiler.py:32]: I am meaning that will your current change break any existing recipes?

Comment (pkufool) [reply icefall/graph_compiler.py:32]: I think it won't break any recipes,  will check it again.

Comment (csukuangfj) [reply icefall/graph_compiler.py:32]: Previously, the repeat attr is always added. But this PR removes it since the arg is always False.

Comment (pkufool) [reply icefall/graph_compiler.py:32]: Yes, you are right, we never tried the phone-based models, but it did support in our code. Will fix it.

Comment (yaozengwei) [reply icefall/graph_compiler.py:32]: > that

Yeah. I also updated the phone-based graph-compiler.

File: egs/librispeech/ASR/conformer_ctc3/train.py [hunk 2]
```diff
@@ -890,6 +890,7 @@ def run(rank, world_size, args):
         graph_compiler = CtcTrainingGraphCompiler(
             lexicon,
             device=device,
+            need_repeat_flag=True if params.delay_penalty > 0 else False,
```
Comment (csukuangfj) [inline egs/librispeech/ASR/conformer_ctc3/train.py:893]:
```suggestion
            need_repeat_flag=params.delay_penalty > 0
```
is simpler

**Review by csukuangfj — APPROVED** [review]

**Pull request merged**